src/panelHeaderWidget.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+
+import { ReactWidget } from '@jupyterlab/apputils';
+
+import {getIdentity} from './utils';
+
+import { Awareness } from 'y-protocols/awareness';
+
+import {CommentsHubIcon, CreateCommentIcon} from './icons';
+
+import {caretDownEmptyIcon} from '@jupyterlab/ui-components'
+
+/**
+ * This type comes from @jupyterlab/apputils/vdom.ts but isn't exported.
+ */
+type ReactRenderElement =
+  | Array<React.ReactElement<any>>
+  | React.ReactElement<any>;
+
+type IdentityProps = {
+  awareness: Awareness | undefined;
+  className: string;
+};
+
+  // Awareness doesn't work unfortunatly 
+function UserIdentity(props: IdentityProps): JSX.Element {
+  const { awareness } = props;
+  const className = props.className || '';
+  return (
+    <div className={className}>
+      {awareness != undefined && getIdentity(awareness).name}
```

Comment:
I think this doesn't update because React doesn't see any changes to `awareness`. So it never re-runs this line and gets a new identity/name.